### PR TITLE
fix: wrong slot starting times for half hour times zones

### LINKED ISSUE #10136: [CAL-2170] Wrong slot starting times for timezones that are not full hours
The slot starting times aren't correct for timezones like Asia/Kolkata (+5:30). Hourly events would start at half-hour slots instead of full-hour slots.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2170](https://linear.app/calcom/issue/CAL-2170/wrong-slot-starting-times-for-timezones-that-are-not-a-full-hour)</sub>

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Fixes that UTC was used to find the time of the first slot. This resulted in wrong slot times for timezones that are not full hours like Asia/Kolkata +5:30.

#### Example: 

availability: 7:30 - 12:00 (+5:30)

Slots for hourly event before: 
Screenshot 2023-07-14 at 09 53 10

Slots for hourly event after: 
Screenshot 2023-07-14 at 09 52 50

Fixes #10136

## Type of change

  - [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

 - Check if tests pass 
 - See if slots start at full hours for and hourly event, availability starting at 7:30 in timezone 5:30

### CODE REVIEW

**Comment by CarinaWolli** [comment]
Looking at the one failed test

File: apps/web/test/lib/getSchedule.test.ts [hunk 1]
```diff
@@ -451,6 +451,8 @@ describe("getSchedule", () => {
         endTime: `${plus2DateString}T18:29:59.999Z`,
         timeZone: Timezones["+5:30"],
       });
+      console.log(JSON.stringify(scheduleForEventWith30Length))
```
Comment (alannnc) [inline apps/web/test/lib/getSchedule.test.ts:454]:
```suggestion
```

**Review by alannnc — DISMISSED** [review]
Code looks good! If test pass we should merge now.

**Review by alannnc — APPROVED** [review]

**Review by zomars — APPROVED** [review]
[cool](https://app.graphite.dev/macros/calcom)

**Pull request merged**